### DISCUSSION TITLE
Input devices like physical buttons/switches

### DISCUSSION THREAD

**Discussion by ThorstenH1972** [catgiggle/OctoPrint-GpioControl#15] [Ideas]
First, thanks for this great plug-in. Works well for me. I like to have single place where I can control everything.

In my use-case I would like also to control GPIOs by physical switches/ buttons because I have no UI in hand.
Would it be possible to select available inputs and assign them to the control path of the outputs (e.g. toggle). Static switches might be problematic to implement and get the initial state, but toggling on an event should be feasible.

Please consider. Would be also volunteer for testing.

Thanks,
Thorsten

**Comment by catgiggle** [comment]
Thanks. I'm glad you are satisfied!

I think it is possible to do exactly what you need. Just connect button directly to your device (eg. relay). You can also connect GPIO in parallel. Thanks to this you will be able to control device (eg. relay) via physical button and GPIO Control too. I'm not sure what is advantage of controlling relay via physical button with GPIO Control in the middle.

But you are not first with your idea. I see there is a need for a software implementation of this. Therefore I will probably implement this functionality, but in the second plugin I'm working on right now. It is similar to GPIO Control but allows more advanced things with GPIO's. This is because I want to keep this plugin as simple as possible.

I'm leaving this discussion open. Maybe there will be more ideas about this functionality.
